pkg/builder/backup_builder.go

```diff
@@ -77,6 +77,9 @@ func (b *BackupBuilder) ObjectMeta(opts ...ObjectMetaOpt) *BackupBuilder {
 
 // FromSchedule sets the Backup's spec and labels from the Schedule template
 func (b *BackupBuilder) FromSchedule(schedule *velerov1api.Schedule) *BackupBuilder {
+	if schedule.Spec.Template.Metadata.Labels != nil {
```

Comment:
This is changing existing behavior and is going to be confusing.

Is the argument that copying the labels from the Schedule is a bad idea?  If so, we should probably just remove the part where it does copy.  If it's OK, why don't we merge the two?  If we want to preserve existing behavior but override (as the code is doing) we should log that we are overriding the Labels copy.

Let's not change the content of the schedule.Labels in here, that's unexpected behavior for the caller.

Reply:
I worked with arush-sal on this and the idea there was that IF you specific .spec.template.metadata.labels (which you can't do before this PR) - then you WANT to make use of the feature this PR provides - which means we switch from copying the schedules labels - to copying the one from the template (which is the method one would expect - which is also used with deployments etc. which have same setup).
How do you feel this is NOT mimicking the way everyones used to decide labels on "created objects" - by Deployments, statefulset etc. ?

Reply:
@dsu-igeek If you have better ways to solve the issue of schedule object labels being copied to backup objects - please share - and we'll gladly listen. In the end we just want "one solution" to be merged - so its possible to use argocd with velero :)
We did all feel that the logic of .spec.templates.metadata (and labels under that) - found in deployments, statefulsets etc. was what we expected when schedule object already HAS .spec.template, but we found that it did not at all implement the expected behavior :( - Thats why we choose this route of action.

Reply:
Let's log what is happening ("Labels (%s) from Schedule.Spec.Template.Metadata.Labels are being applied to Backup %s", "Labels (%s) from Schedule are being applied to Backup %s" or something like that.

Please change the logic so that the in-memory schedule.Labels is not modified.  schedule is coming in as a pointer so the code is modifying the structure passed in.

Reply:
@dsu-igeek the above has been addressed. Please have a look and let me know.